[2024-11-22T18:59:00.292+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-22T18:59:00.347+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-22T18:57:00+00:00 [queued]>
[2024-11-22T18:59:00.364+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-22T18:57:00+00:00 [queued]>
[2024-11-22T18:59:00.365+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2024-11-22T18:59:00.423+0000] {taskinstance.py:2888} INFO - Executing <Task(BranchPythonOperator): process_transaction> on 2024-11-22 18:57:00+00:00
[2024-11-22T18:59:00.435+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=839) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-22T18:59:00.438+0000] {standard_task_runner.py:72} INFO - Started process 842 to run task
[2024-11-22T18:59:00.443+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'fraud_detection', 'process_transaction', 'scheduled__2024-11-22T18:57:00+00:00', '--job-id', '452', '--raw', '--subdir', 'DAGS_FOLDER/fraud_detection_dag.py', '--cfg-path', '/tmp/tmpweqb0ofv']
[2024-11-22T18:59:00.447+0000] {standard_task_runner.py:105} INFO - Job 452: Subtask process_transaction
[2024-11-22T18:59:00.536+0000] {task_command.py:467} INFO - Running <TaskInstance: fraud_detection.process_transaction scheduled__2024-11-22T18:57:00+00:00 [running]> on host c6af8dbaf1e5
[2024-11-22T18:59:00.696+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='fraud_team' AIRFLOW_CTX_DAG_ID='fraud_detection' AIRFLOW_CTX_TASK_ID='process_transaction' AIRFLOW_CTX_EXECUTION_DATE='2024-11-22T18:57:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-22T18:57:00+00:00'
[2024-11-22T18:59:00.698+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-22T18:59:00.712+0000] {baseoperator.py:405} WARNING - BranchPythonOperator.execute cannot be called outside TaskInstance!
[2024-11-22T18:59:00.734+0000] {fraud_detection_dag.py:77} INFO - Données récupérées depuis XCom: {"columns":["cc_num","merchant","category","amt","first","last","gender","street","city","state","zip","lat","long","city_pop","job","dob","trans_num","merch_lat","merch_long","is_fraud","current_time"],"index":[191872],"data":[[180048185037117,"fraud_Schuppe, Nolan and Hoeger","shopping_pos",178.89,"Mary","Wall","F","2481 Mills Lock","Plainfield","NJ",7060,40.6152,-74.415,71485,"Leisure centre manager","1974-07-19","f2f12e30b7539afcba2e84d21054d7b8",40.344971,-74.880225,0,1732301938392]]}
[2024-11-22T18:59:00.735+0000] {fraud_detection_dag.py:85} INFO - Données JSON parsées avec succès
[2024-11-22T18:59:00.740+0000] {fraud_detection_dag.py:89} INFO - DataFrame créé avec 1 lignes
[2024-11-22T18:59:00.742+0000] {fraud_detection_dag.py:93} INFO - Colonnes renommées avec succès
[2024-11-22T18:59:00.743+0000] {fraud_detection_dag.py:94} INFO - Colonnes disponibles: ['cc_num', 'merchant', 'category', 'amt', 'first', 'last', 'gender', 'street', 'city', 'state', 'zip', 'lat', 'long', 'city_pop', 'job', 'dob', 'trans_num', 'merch_lat', 'merch_long', 'is_fraud', 'trans_date_trans_time']
[2024-11-22T18:59:00.746+0000] {fraud_detection_dag.py:98} INFO - Conversion de trans_date_trans_time en unix_time réussie
[2024-11-22T18:59:00.790+0000] {fraud_detection_dag.py:107} INFO - Transformation des données...
[2024-11-22T18:59:01.941+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-22T18:59:01.967+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-22T18:59:01.992+0000] {fraud_detection_dag.py:119} INFO - Prediction de fraude: NORMAL, Probabilité: 33.88%
[2024-11-22T18:59:02.189+0000] {fraud_detection_dag.py:125} INFO - Prédiction: NORMAL (Probabilité: 33.88%)
[2024-11-22T18:59:02.190+0000] {python.py:240} INFO - Done. Returned value was: notify_normal
[2024-11-22T18:59:02.191+0000] {branch.py:38} INFO - Branch into notify_normal
[2024-11-22T18:59:02.192+0000] {skipmixin.py:230} INFO - Following branch ('notify_normal',)
[2024-11-22T18:59:02.220+0000] {skipmixin.py:278} INFO - Skipping tasks [('skip_processing', -1), ('notify_fraud', -1)]
[2024-11-22T18:59:02.281+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-22T18:59:02.282+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=fraud_detection, task_id=process_transaction, run_id=scheduled__2024-11-22T18:57:00+00:00, execution_date=20241122T185700, start_date=20241122T185900, end_date=20241122T185902
[2024-11-22T18:59:02.442+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-11-22T18:59:02.525+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-11-22T18:59:02.530+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
